### PR TITLE
Core: update hooks

### PR DESCRIPTION
**Pull request by Daztek**
Hook order args could use a look, went with Early for the POS ones, wondering if Final is better or not.

### CODE REVIEW

File: NWNXLib/Services/PerObjectStorage/PerObjectStorage.hpp [hunk 1]
```diff
@@ -36,12 +34,12 @@ class PerObjectStorage
...
-    static void CNWSPlayer__EatTURD_hook(bool, CNWSPlayer* thisPtr, CNWSPlayerTURD* pTURD);
-    static void CNWSPlayer__DropTURD_hook(bool, CNWSPlayer* thisPtr);
-    static void CNWSUUID__SaveToGff_hook(bool, CNWSUUID* pThis, CResGFF* pRes, CResStruct* pStruct);
-    static void CNWSUUID__LoadFromGff_hook(bool, CNWSUUID* pThis, CResGFF* pRes, CResStruct* pStruct);
+    static void CNWSObject__CNWSObjectDtor__0_hook(CNWSObject* thisPtr);
+    static void CNWSArea__CNWSAreaDtor__0_hook(CNWSArea* thisPtr);
+    static void CNWSPlayer__EatTURD_hook(CNWSPlayer* thisPtr, CNWSPlayerTURD* pTURD);
+    static void CNWSPlayer__DropTURD_hook(CNWSPlayer* thisPtr);
+    static void CNWSUUID__SaveToGff_hook(CNWSUUID* pThis, CResGFF* pRes, CResStruct* pStruct);
+    static bool CNWSUUID__LoadFromGff_hook(CNWSUUID* pThis, CResGFF* pRes, CResStruct* pStruct);
```
Comment (mtijanic) [inline NWNXLib/Services/PerObjectStorage/PerObjectStorage.hpp:42]: bool is not the same as BOOL. BOOL is an int32_t actually

Comment (Daztek) [reply NWNXLib/Services/PerObjectStorage/PerObjectStorage.hpp:42]: It's actually a `bool` in this case: https://github.com/nwnxee/unified/blob/master/NWNXLib/API/API/CNWSUUID.hpp#L29

Comment (mtijanic) [reply NWNXLib/Services/PerObjectStorage/PerObjectStorage.hpp:42]: Oh, wow, yeah. That's niv-code if I ever saw one :)

File: Core/NWNXCore.cpp [hunk 2]
```diff
@@ -170,59 +170,61 @@ void NWNXCore::ConfigureLogLevel(const std::string& plugin, const NWNXLib::Servi
...
+    m_vmTagEffectHook      = m_services->m_hooks->Hook(API::Functions::_ZN25CNWVirtualMachineCommands23ExecuteCommandTagEffectEii, (void*)&TagEffectHandler, Hooking::Order::Final);
+    m_vmTagItemProperyHook = m_services->m_hooks->Hook(API::Functions::_ZN25CNWVirtualMachineCommands29ExecuteCommandTagItemPropertyEii, (void*)&TagItemPropertyHandler, Hooking::Order::Final);
+    m_vmPlaySoundHook      = m_services->m_hooks->Hook(API::Functions::_ZN25CNWVirtualMachineCommands23ExecuteCommandPlaySoundEii, (void*)&PlaySoundHandler, Hooking::Order::Final);
+
+    m_destroyServerHook    = m_services->m_hooks->Hook(API::Functions::_ZN11CAppManager13DestroyServerEv, (void*)&DestroyServerHandler, Hooking::Order::Final);
+    m_mainLoopInternalHook = m_services->m_hooks->Hook(API::Functions::_ZN21CServerExoAppInternal8MainLoopEv, (void*)&MainLoopInternalHandler, Hooking::Order::Final);
+
+    m_posObjectDtorHook      = m_services->m_hooks->Hook(API::Functions::_ZN10CNWSObjectD1Ev, (void*)&Services::PerObjectStorage::CNWSObject__CNWSObjectDtor__0_hook, Hooking::Order::Early);
```
Comment (mtijanic) [inline Core/NWNXCore.cpp:182]: I would go with VeryEarly or Earliest for POS

File: Core/NWNXCore.cpp [hunk 3]
```diff
@@ -170,59 +170,61 @@ void NWNXCore::ConfigureLogLevel(const std::string& plugin, const NWNXLib::Servi
...
+                API::Functions::_ZN15CVirtualMachine14RunScriptChunkERK10CExoStringjii,
+                (void*)+[](CVirtualMachine *pVirtualMachine, const CExoString& sScriptChunk, ObjectID oid, int32_t bOidValid, int32_t bWrapIntoMain) -> int32_t
                 {
-                    g_core->m_ScriptChunkRecursion += before ? +1 : -1;
-                });
+                    g_core->m_ScriptChunkRecursion += 1;
+                    auto retVal = runScriptChunkHook->CallOriginal<int32_t>(pVirtualMachine, sScriptChunk, oid, bOidValid, bWrapIntoMain);
+                    g_core->m_ScriptChunkRecursion -= 1;
+                    return retVal;
+                }, Hooking::Order::Early);
```
Comment (mtijanic) [inline Core/NWNXCore.cpp:227]: Again VeryEarly, or Earliest.

**Review by mtijanic — COMMENTED** [review]
I'll clean up the ugly reaching into g_core from services at some point. This is good for now.

**Pull request merged**